### PR TITLE
fix null _server: the private class field _server is never inited.

### PR DESCRIPTION
**Pull request by mw66**
-- fix null _server: the private class field _server is never inited.
-- http://grpc.github.io/grpc/core/grpc_8h.html#aceedc7178f15ebef5f518ca180938a20

GRPCAPI void grpc_shutdown_blocking	(	void 		)	
DEPRECATED.

Recommend to use grpc_shutdown only

-- fix DMD64 D Compiler v2.104.0 new compiling error

-- life is too short to debug dlang built-in AA to right, let's just use HashMap from emsi_containers.
Fixed the problem found here:
https://forum.dlang.org/thread/duetqujuoceancqtjlar@forum.dlang.org

### CODE REVIEW

**Comment by hatf0** [comment]
Hey there,

Just as a little heads up - I haven't touched this source code in forever (I wrote this when I was 17) & have since left the D language community entirely. I'd be happy to transfer over this repository (and all other related ones, including the compiler / etc) to your control, as well as the Dub package. I'm glad to see it finally get some real practical use though! I spent an utterly absurd amount of time fine-tuning this.

File: source/grpc/common/byte_buffer.d [hunk 1]
```diff
@@ -96,7 +96,8 @@ struct ByteBuffer {
             reader = reader.init;
         }
         
-        return slice_to_type!(ubyte[])(slice);
+	ubyte[] r = slice_to_type!(ubyte[])(slice);
```
Comment (hatf0) [inline source/grpc/common/byte_buffer.d:99]: Uhhh... is there any reason why this type can't be inferred by the compiler? Did something related to alias this break recently?

File: source/grpc/server/package.d [hunk 2]
```diff
@@ -83,18 +83,23 @@ class Server
...
+    import std.stdio;
     void wait() {
         foreach(service; services.keys) {
             services[service].kickstart();
         }
 
         while (atomicLoad(_run)) {
+		writeln(services.keys);  // print out some wrong dict value of other (un-related) parts of the program
             
+	/* https://forum.dlang.org/thread/duetqujuoceancqtjlar@forum.dlang.org
```
Comment (hatf0) [inline source/grpc/server/package.d:99]: Oh ... boy. Iterating over an AA causing a segmentation fault. This is one of many reasons why I stopped writing D

Comment (mw66) [reply source/grpc/server/package.d:99]: I think there are some bugs in the builtin AA implementation, as demo-ed in my multi-threaded program (maybe also related to GC).

This AA is corrupted at some point for no reason.

File: source/grpc/common/byte_buffer.d (hunk 1 above)
Comment (mw66) [reply source/grpc/common/byte_buffer.d:99]: Not about the type.

The new compiler complains some local storage is escaped.

With explicit local var, it's fixed.

**Comment by mw66** [comment]
> Hey there,
> 
> Just as a little heads up - I haven't touched this source code in forever (I wrote this when I was 17) & have since left the D language community entirely. I'd be happy to transfer over this repository (and all other related ones, including the compiler / etc) to your control, as well as the Dub package. I'm glad to see it finally get some real practical use though! I spent an utterly absurd amount of time fine-tuning this.

Thanks for your contribution.

You can transfer to me, and I will transfer to the D-community one day.

I think it's a pretty decent package, better than the other grpc-dlang package that I have tried.

If you have time, maybe you can answer some of my questions, or add some more doc to make the transfer more easy to me.

Thanks!

**Comment by hatf0** [comment]
> 
> Thanks for your contribution.
> 
> You can transfer to me, and I will transfer to the D-community one day.
> 
> I think it's a pretty decent package, better than the other grpc-dlang package that I have tried.
> 
> If you have time, maybe you can answer some of my questions, or add some more doc to make the transfer more easy to me.
> 
> Thanks!

Feel free to send any questions my way.

File: source/grpc/server/package.d (hunk 2 above)
Comment (hatf0) [reply source/grpc/server/package.d:99]: Yeah ... When I was writing D professionally we seldom touched the standard library.

File: source/grpc/common/byte_buffer.d (hunk 1 above)
Comment (hatf0) [reply source/grpc/common/byte_buffer.d:99]: WTF? Did something break with `alias this` / inout recently?

**Comment by hatf0** [comment]
Cherry-picked parts of this PR into master - tagged as v0.0.3

**Pull request closed without merging**